crates/shadowsocks/src/relay/udprelay/proxy_socket.rs

```diff
@@ -72,7 +75,7 @@ pub type ProxySocketResult<T> = Result<T, ProxySocketError>;
 #[derive(Debug)]
 pub struct ProxySocket {
     socket_type: UdpSocketType,
-    socket: ShadowUdpSocket,
+    io: Box<dyn DatagramTransport>,
```

Comment:
Why not just make `io` to be a generic parameter

```rust
pub struct ProxySocket<S>
    where S: DatagramTransport
{
    socket: S
}
```

Reply:
As mentioned above, `S` could be `S: DatagramTransport + Send + Sync + Debug`

Reply:
i think that'd make the ProxySocket to be a generic, which can propgate to the entire call chain.

but I prefer it too, which would be static dispatch instead of dynamic